domain/secret/state/state.go

```diff
@@ -907,6 +983,67 @@ WHERE  mso.label = $M.label
 	return coresecrets.ParseURI(dbSecrets[0].ID)
 }
 
+// GetURIByConsumerLabel looks up the secret URI using the label previously registered by the specified unit,
+// returning an error satisfying [secreterrors.SecretNotFound] if there's no corresponding URI.
+// If the unit does not exist, an error satisfying [uniterrors.NotFound] is returned.
+func (st State) GetURIByConsumerLabel(ctx context.Context, label string, unitName string) (*coresecrets.URI, error) {
+	if label == "" {
+		return nil, errors.NotValidf("empty secret label")
+	}
+
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	query := `	
+SELECT secret_id AS &secretUnitConsumer.secret_id
+FROM   secret_unit_consumer suc
+WHERE  suc.label = $secretUnitConsumer.label
+AND    suc.unit_uuid = $secretUnitConsumer.unit_uuid
+`
+
+	queryStmt, err := st.Prepare(query, secretUnitConsumer{}, sqlair.M{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	selectUnitUUID := `select &M.uuid FROM unit WHERE unit_id=$M.unit_id`
+	selectUnitUUIDStmt, err := st.Prepare(selectUnitUUID, sqlair.M{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var dbConsumers []secretUnitConsumer
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		result := sqlair.M{}
+		err = tx.Query(ctx, selectUnitUUIDStmt, sqlair.M{"unit_id": unitName}).Get(&result)
+		if err != nil {
+			if errors.Is(err, sqlair.ErrNoRows) {
+				return uniterrors.NotFound
```

Comment:
Include the unitName in the error message.

Reply:
fixed all of them